ui/components/multichain/pages/gator-permissions/components/review-gator-permission-item.tsx

```diff
@@ -115,46 +170,34 @@ export const ReviewGatorPermissionItem = ({
   const permissionType = permissionResponse.permission.type;
   const permissionContext = permissionResponse.context;
   const permissionAccount = permissionResponse.address || '0x';
+  const justification = permissionResponse.permission.data.justification as
+    | string
+    | undefined;
   const tokenAddress = permissionResponse.permission.data.tokenAddress as
-    | Hex
+    | string
     | undefined;
 
   const [isExpanded, setIsExpanded] = useState(false);
-  const tokensByChain = useSelector(selectERC20TokensByChain);
-  const nativeTokenMetadata = useSelector((state) =>
-    getNativeTokenInfo(state, chainId),
-  ) as TokenMetadata;
   const pendingRevocations = useSelector(getPendingRevocations);
+  const internalAccount = useSelector((state) =>
+    getInternalAccountByAddress(state, permissionAccount),
+  );
 
-  const tokenMetadata: TokenMetadata = useMemo(() => {
-    if (tokenAddress) {
-      const tokenListForChain = tokensByChain?.[chainId]?.data || {};
-      const foundTokenMetadata = getTokenMetadata(
-        tokenAddress,
-        tokenListForChain,
-      );
-      if (foundTokenMetadata) {
-        return {
-          symbol: foundTokenMetadata.symbol || 'Unknown Token',
-          decimals: foundTokenMetadata.decimals || 18,
-          name: foundTokenMetadata.name || 'Unknown Token',
-        };
-      }
-      console.warn(
-        `Token metadata not found for address: ${tokenAddress} for chain: ${chainId}`,
-      );
-      return {
-        symbol: 'Unknown Token',
-        decimals: null,
-        name: 'Unknown Token',
-      };
-    }
-    return {
-      symbol: nativeTokenMetadata.symbol,
-      decimals: nativeTokenMetadata.decimals,
-      name: nativeTokenMetadata.name,
-    };
-  }, [tokensByChain, chainId, tokenAddress, nativeTokenMetadata]);
+  const truncatedAddress = useMemo(
+    () => shortenAddress(permissionAccount),
+    [permissionAccount],
+  );
+
+  const accountText = useMemo(() => {
+    return internalAccount?.metadata?.name || truncatedAddress;
+  }, [internalAccount, truncatedAddress]);
```

Comment:
It looks like `truncatedAddress` isn't used anywhere else - could we put that inside the accountText `useMemo` to improve readability?
```suggestion
  const accountText = useMemo(() => {
    return internalAccount?.metadata?.name || shortenAddress(permissionAccount);
  }, [internalAccount, permissionAccount]);
```